eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -1073,6 +1084,22 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder, val
 
     case Event(BITCOIN_FUNDING_TIMEOUT, d: DATA_CLOSING) => handleFundingTimeout(d)
 
+    case Event(w: WatchFundingConfirmedTriggered, d: DATA_CLOSING) =>
+      d.alternativeCommitments.find(_.commitInput.outPoint.txid == w.tx.txid) match {
+        case Some(alternativeCommitments) =>
+          log.info("an alternative funding tx with txid={} got confirmed", w.tx.txid)
+          watchFundingTx(alternativeCommitments)
+          context.system.eventStream.publish(TransactionConfirmed(d.channelId, remoteNodeId, w.tx))
+          val commitTx = alternativeCommitments.fullySignedLocalCommitTx(keyManager).tx
+          val localCommitPublished = Closing.LocalClose.claimCommitTxOutputs(keyManager, alternativeCommitments, commitTx, nodeParams.currentBlockHeight, nodeParams.onChainFeeConf)
+          stay() using d.copy(commitments = alternativeCommitments, localCommitPublished = Some(localCommitPublished)) storing() calling doPublish(localCommitPublished, alternativeCommitments)
```

Comment:
That's not correct, we must forget about all `*Proposed`/`*Published`, they have been double spent. If we are logical we should also forget the alternate commitments. I think the best is to create a brand new `DATA_CLOSING`. It makes sense because we are switching from a completely separate commitment tree.

Reply:
Agreed, that makes more sense.

Reply:
Done in https://github.com/ACINQ/eclair/pull/2274/commits/7b7bfac5bb44092cac0a7439723abff63ae4f356